### PR TITLE
Added basic functionality for parameter store

### PR DESCRIPTION
**Pull request by chefbob**
Signed-off-by: chefbob

### Description
Created initial code to support AWS parameter store.  There is code for get, create and delete.

### Issues Resolved
Issue 327

### Check List

- [ ] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [x] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

**Comment by chefbob** [comment]
Still need to do readme.  Also need to figure out how to assign results of get to a variable so that the data can be used by people who include the cookbook.  The data may contain secrets so we want to give the user the choice of putting it in templates, attributes, files, etc.
Once get is working we'll need to write test cases for it.

File: resources/ssm_parameter_store.rb [hunk 1]
```diff
@@ -0,0 +1,124 @@
+property :name,                        String, required: true
```
Comment (tas50) [inline resources/ssm_parameter_store.rb]: You don't want to define the name property. Chef defines it for you already and it actually has a few more options out of the box

File: resources/ssm_parameter_store.rb [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+property :description,                 String
+property :value,                       String, required: true
+property :type,                        String, required: true
+property :key_id,              		   String
+property :overwrite,                   [true, false], default: true
+property :with_decryption,             [true, false], default: false
```
Comment (tas50) [inline resources/ssm_parameter_store.rb]: There's some extra spaces here

File: resources/ssm_parameter_store.rb [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    request = {
+      name: name,
+	  with_decryption: with_decryption,
+    }
+    response = ssm_client.get_parameter(request)
+    Chef::Log.debug "Get parameter #{name}"
+end
+
+action :create do
+	if (write_parameter)
```
Comment (tas50) [inline resources/ssm_parameter_store.rb]: No need for () here

File: resources/ssm_parameter_store.rb [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+  def with_decryption
+    @with_decryption ||= new_resource.with_decryption
+  end
+  
+  def allowed_pattern
+    @allowed_pattern ||= new_resource.allowed_pattern
+  end
+
+  def write_parameter
+  # If the paremeter doesn't exist or one of the values has changed and overwrite
```
Comment (tas50) [inline resources/ssm_parameter_store.rb]: paremeter -> parameter

**Comment by tas50** [comment]
You'll wan to run Cookstyle against this since there's a pile of warning. Looks prety nice though. This might be a good way for us to store state information within the cookbook which would resolve a lot of the longstanding issues. As for using it within the cookbook you'd probably want to write some sort of library helper to handle getting the values.

**Pull request merged**